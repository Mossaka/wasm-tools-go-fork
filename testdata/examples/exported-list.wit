package examples:exported-list;

interface command {
	bytes: func() -> list<u8>;
}

world exports {
	export command;
}
