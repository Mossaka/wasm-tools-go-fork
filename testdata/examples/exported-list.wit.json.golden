&wit.Resolve{
    Worlds: {
        &wit.World{
            _typeOwner: wit._typeOwner{},
            Name:       "exports",
            Imports:    {},
            Exports:    {
                "interface-0": &wit.Interface{
                    _typeOwner: wit._typeOwner{},
                    _worldItem: wit._worldItem{},
                    Name:       &"command",
                    TypeDefs:   {},
                    Functions:  {
                        "bytes": &wit.Function{
                            _worldItem: wit._worldItem{},
                            Name:       "bytes",
                            Kind:       &wit.Freestanding{},
                            Params:     nil,
                            Results:    {
                                {
                                    Name: "",
                                    Type: &wit.TypeDef{
                                        _type:      wit._type{},
                                        _worldItem: wit._worldItem{},
                                        Name:       (*string)(nil),
                                        Kind:       &wit.List{
                                            _typeDefKind: wit._typeDefKind{},
                                            Type:         wit.U8{},
                                        },
                                        Owner: nil,
                                        Docs:  wit.Docs{},
                                    },
                                },
                            },
                            Docs: wit.Docs{},
                        },
                    },
                    Package: &wit.Package{
                        Name: wit.Ident{
                            Namespace: "examples",
                            Package:   "exported-list",
                            Extension: "",
                            Version:   (*semver.Version)(nil),
                        },
                        Interfaces: {
                            "command": &wit.Interface{(CYCLIC REFERENCE)},
                        },
                        Worlds: {
                            "exports": &wit.World{(CYCLIC REFERENCE)},
                        },
                        Docs: wit.Docs{},
                    },
                    Docs: wit.Docs{},
                },
            },
            Package: &wit.Package{
                Name: wit.Ident{
                    Namespace: "examples",
                    Package:   "exported-list",
                    Extension: "",
                    Version:   (*semver.Version)(nil),
                },
                Interfaces: {
                    "command": &wit.Interface{
                        _typeOwner: wit._typeOwner{},
                        _worldItem: wit._worldItem{},
                        Name:       &"command",
                        TypeDefs:   {},
                        Functions:  {
                            "bytes": &wit.Function{
                                _worldItem: wit._worldItem{},
                                Name:       "bytes",
                                Kind:       &wit.Freestanding{},
                                Params:     nil,
                                Results:    {
                                    {
                                        Name: "",
                                        Type: &wit.TypeDef{
                                            _type:      wit._type{},
                                            _worldItem: wit._worldItem{},
                                            Name:       (*string)(nil),
                                            Kind:       &wit.List{
                                                _typeDefKind: wit._typeDefKind{},
                                                Type:         wit.U8{},
                                            },
                                            Owner: nil,
                                            Docs:  wit.Docs{},
                                        },
                                    },
                                },
                                Docs: wit.Docs{},
                            },
                        },
                        Package: &wit.Package{(CYCLIC REFERENCE)},
                        Docs:    wit.Docs{},
                    },
                },
                Worlds: {
                    "exports": &wit.World{(CYCLIC REFERENCE)},
                },
                Docs: wit.Docs{},
            },
            Docs: wit.Docs{},
        },
    },
    Interfaces: {
        &wit.Interface{
            _typeOwner: wit._typeOwner{},
            _worldItem: wit._worldItem{},
            Name:       &"command",
            TypeDefs:   {},
            Functions:  {
                "bytes": &wit.Function{
                    _worldItem: wit._worldItem{},
                    Name:       "bytes",
                    Kind:       &wit.Freestanding{},
                    Params:     nil,
                    Results:    {
                        {
                            Name: "",
                            Type: &wit.TypeDef{
                                _type:      wit._type{},
                                _worldItem: wit._worldItem{},
                                Name:       (*string)(nil),
                                Kind:       &wit.List{
                                    _typeDefKind: wit._typeDefKind{},
                                    Type:         wit.U8{},
                                },
                                Owner: nil,
                                Docs:  wit.Docs{},
                            },
                        },
                    },
                    Docs: wit.Docs{},
                },
            },
            Package: &wit.Package{
                Name: wit.Ident{
                    Namespace: "examples",
                    Package:   "exported-list",
                    Extension: "",
                    Version:   (*semver.Version)(nil),
                },
                Interfaces: {
                    "command": &wit.Interface{(CYCLIC REFERENCE)},
                },
                Worlds: {
                    "exports": &wit.World{(CYCLIC REFERENCE)},
                },
                Docs: wit.Docs{},
            },
            Docs: wit.Docs{},
        },
    },
    TypeDefs: {
        &wit.TypeDef{
            _type:      wit._type{},
            _worldItem: wit._worldItem{},
            Name:       (*string)(nil),
            Kind:       &wit.List{
                _typeDefKind: wit._typeDefKind{},
                Type:         wit.U8{},
            },
            Owner: nil,
            Docs:  wit.Docs{},
        },
    },
    Packages: {
        &wit.Package{
            Name: wit.Ident{
                Namespace: "examples",
                Package:   "exported-list",
                Extension: "",
                Version:   (*semver.Version)(nil),
            },
            Interfaces: {
                "command": &wit.Interface{(CYCLIC REFERENCE)},
            },
            Worlds: {
                "exports": &wit.World{(CYCLIC REFERENCE)},
            },
            Docs: wit.Docs{},
        },
    },
}